tpot/base.py

```diff
@@ -391,6 +396,55 @@ def _setup_config(self, config_dict):
         else:
             self._config_dict = self.default_config_dict
 
+    def _setup_preprocess_config(self, config_dict):
+        if self.template is None:
+            self.template = 'RandomTree'
+        elif 'PreprocessTransformer' in self.template:
+            return
+        if config_dict:
+            # check for valid keys...
+            preprocess_keys = ['text_columns', 'numeric_columns', 'categorical_columns', 'impute']
+            column_transform_keys = ['text_columns', 'numeric_columns', 'categorical_columns', 'text_transformer', 'categorical_transformer']
+            check_keys_ = []
+            column_transform_ = []
+            for k in config_dict.keys():
+                if k not in preprocess_keys:
+                    check_keys_.append(k)
+                if k in column_transform_keys:
+                    column_transform_.append(k)
+
+            if len(check_keys_) > 0:
+                raise Exception(
+                    'It appears preprocessing configuration contains invalid keys: {}.'
+                    'Valid keys are text_columns, numeric_columns,'
+                    'categorical_columns, impute.'.format(', '.join(check_keys_))
+                )
+
+            # override some settings...
+            if config_dict.get('impute', None) is False:
```

Comment:
This line deals with #836 - might be overloading this PR and might be an item for later?

Reply:
I think it is more related to #889. I think we need add imputation into `config_dict` too. We may allow TPOT skip imputation if the pipeline only has XGBClassifier or XGBRegressor.